test/selector_assertions_test.rb

```diff
@@ -312,6 +312,69 @@ def test_body_class_can_be_tested_with_html
     assert_select '.foo'
   end
 
+
+  #
+  # Test within
+  #
+
+  def test_assert_dom_nested_in_within_with_css
+    render_html %Q{<aside>other</aside><div id="1"><span>foo</span></div><div id="2"><span>bar</span></div>}
+    within "div[id=?]", "1", count: 1 do
+      assert_dom "div", text: "foo", count: 1
+      assert_dom "span", text: "foo", count: 1
+      assert_dom "div", text: "bar", count: 0
+      assert_dom "span", text: "bar", count: 0
+      assert_dom "aside", text: "other", count: 0
+    end
+    assert_dom "div", text: "foo", count: 1
+    assert_dom "div", text: "bar", count: 1
+    assert_dom "aside", text: "other", count: 1
+  end
+
+  def test_assert_dom_nested_in_within_with_node
+    render_html %Q{<aside>other</aside><div id="1"><span>foo</span></div><div id="2"><span>bar</span></div>}
+    within document_root_element.at_css("div") do
+      assert_dom "div", text: "foo", count: 1
+      assert_dom "span", text: "foo", count: 1
+      assert_dom "div", text: "bar", count: 0
+      assert_dom "span", text: "bar", count: 0
+      assert_dom "aside", text: "other", count: 0
+    end
+    assert_dom "div", text: "foo", count: 1
+    assert_dom "div", text: "bar", count: 1
+    assert_dom "aside", text: "other", count: 1
+  end
+
+  def test_assert_dom_nested_in_within_with_nodeset
+    render_html %Q{<aside>other</aside><div id="1"><span>foo</span></div><div id="2"><span>bar</span></div>}
+    within css_select("div") do
+      assert_dom "div", text: "foo", count: 1
+      assert_dom "span", text: "foo", count: 1
+      assert_dom "div", text: "bar", count: 1
+      assert_dom "span", text: "bar", count: 1
+      assert_dom "aside", text: "other", count: 0
+    end
+    assert_dom "div", text: "foo", count: 1
+    assert_dom "div", text: "bar", count: 1
+    assert_dom "aside", text: "other", count: 1
+  end
+
+  def test_missing_selector_passed_to_within
+    render_html %Q{<div></div>}
+
+    assert_raises Minitest::Assertion do
+      within("#missing") {}
+    end
+  end
+
+  def test_nil_selection_passed_to_within
+    render_html %Q{<div></div>}
+
+    assert_raises ArgumentError, /Node, NodeSet, or selector/ do
```

Comment:
iirc, the second argument here is going to be part of minitests failure message if the `assert_raises` fails, not adding an extra assertion on the message.

You'd have to do

```ruby
error = assert_raises…
assert_match //, error.message
```

Reply:
Good catch, thanks!